src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -772,11 +772,21 @@ ISymbol ITypeSymbol.FindImplementationForInterfaceMember(ISymbol interfaceMember
         /// </summary>
         bool ITypeSymbol.IsTupleType => this.IsTupleType;
 
+        public string ToDisplayString(CodeAnalysis.NullableAnnotation topLevelNullability, SymbolDisplayFormat format = null)
+        {
+            return SymbolDisplay.ToDisplayString(this, topLevelNullability, format);
+        }
```

Comment:
can you clarify these new ToDisplayStrings?  Do callers have to know to call them in order to get proper nullability behavior?  How does that affect our own code and the ecosystem of code out there that is using the existing ToDisplayString.

Reply:
Just to write up some details for the current state for anybody else following along: yes, you have to know them or else you lose top-level nullability. This was a known caveat of this design and we're reconsidering it in light of this PR; we're going to continue with this PR in this form since at least it means we're fixing one feature, it unblocks moving other features to the current design, and also if we _do_ change the API it means cleaning things up is easier too -- we can find refs on the method we're deleting and fix it up.